### ISSUE TITLE
Cross entropy Loss : Ignore list of indexes 

### ISSUE THREAD

**Issue by jibrilfrej** [pytorch/pytorch#12601]
## 🚀 Feature
Generalize ignore_index to list of indexes instead of single index for the cross entropy loss.

## Motivation

I would like to ignore several classes during training.

## Alternatives

Right now I am assigning a weight = 0 to the classes I want to ignore during training. 
Is it equivalent to ignore_index over the list of indexes I want to ignore?

**Comment by soumith** [comment]
while this is a "nice to have", it's not popular enough that we want to complicate the function even more in it's implementation.
You can do some pre-processing on your `targets` to get to the same place, or as you mentioned, use the weight=0 trick. In fact, `weight` is provided for this precise reason.
